### PR TITLE
AO3-6625 Replace whitespace in download filenames with underscores

### PR DESCRIPTION
**Pull request by sarken**
## Issue

https://otwarchive.atlassian.net/browse/AO3-6625

## Purpose

Replaces the whitespace in download filenames with underscores to ensure the Calibre-generated table of contents works on EPUBs for Kindle. (Refer to Jira for more info on why this should fix it despite seeming unconnected.)

## Testing Instructions

Refer to Jira.

## Credit

Sarken, she/her

### CODE REVIEW

File: spec/models/download_spec.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@
       expect(Download.new(work).file_name).to eq("123456789-123456789-1234")
 
       work.title = "123456789 123456789 123456789"
-      expect(Download.new(work).file_name).to eq("123456789 123456789")
+      expect(Download.new(work).file_name).to eq("123456789_123456789")
     end
```
Comment (ceithir) [inline spec/models/download_spec.rb:57]: Maybe an additional test for the edge case of a work title having several consecutive spaces like `A         mystery`?

File: app/models/download.rb [hunk 2]
```diff
@@ -51,10 +51,15 @@ def file_type_from_mime(mime)
...
+  # The base name of the file (e.g., "War_and_Peace")
   def file_name
     name = clean(work.title)
-    name += " Work #{work.id}" if name.length < 3
+    extender = "Work_#{work.id}"
+    if name.length.zero?
+      name = extender
+    elsif name.length < 3
+      name += "_#{extender}"
+    end
```
Comment (ceithir) [inline app/models/download.rb:62]: My JS-addled immediately thought of "simplifying" that using `join`:
```rb
name = [name, "Work_#{work.id}"].compact_blank.join("_") if name.length < 3
```
But not exactly sure this is indeed better.

**Review by ceithir — APPROVED** [review]

**Review by brianjaustin — APPROVED** [review]

**Review by neuroalien — APPROVED** [review]
Approving on the strength of the two existing approvals!
